### PR TITLE
Small adjustments to the READMEs

### PR DESCRIPTION
**Pull request by lilidworkin**
## Description of the change

Self explanatory, just updating some copy. Also working on corresponding updates to the backend and frontend READMEs in recidiviz-data.

## Type of change

> All pull requests must have at least one of the following labels applied (otherwise the PR will fail):

| Label                       	| Description                                                                                               	|
|-----------------------------	|-----------------------------------------------------------------------------------------------------------	|
| Type: Bug                   	| non-breaking change that fixes an issue                                                                   	|
| Type: Feature               	| non-breaking change that adds functionality                                                               	|
| Type: Breaking Change       	| fix or feature that would cause existing functionality to not work as expected                            	|
| Type: Non-breaking refactor 	| change addresses some tech debt item or prepares for a later change, but does not change functionality    	|
| Type: Configuration Change  	| adjusts configuration to achieve some end related to functionality, development, performance, or security 	|
| Type: Dependency Upgrade      | upgrades a project dependency - these changes are not included in release notes                             |

### Development

**This box MUST be checked by the submitter prior to merging**:
- [X] **Double- and triple-checked that there is no Personally Identifiable Information (PII) being mistakenly added in this pull request**

These boxes should be checked by the submitter prior to merging:
- [ ] Tests have been written to cover the code changed/added as part of this pull request

### Code review

These boxes should be checked by reviewers prior to merging:

- [ ] This pull request has a descriptive title and information useful to a reviewer
- [ ] This pull request has been moved out of a Draft state, has no "Work In Progress" label, and has assigned reviewers
- [ ] Potential security implications or infrastructural changes have been considered, if relevant

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
-# justice-counts
+# Justice Counts
 
-Technical infrastructure for the Justice Counts initiative
+This repository contains the technical infrastructure for the Justice Counts initiative. This consists of the following components:
+
+- **Publisher** -- a tool that allows agencies to report Justice Counts metrics. The frontend of this application lives in the `/publisher` directory of this repo. The backend of this application lives in the [`recidiviz-data`](https://github.com/Recidiviz/recidiviz-data) repo.
```
Comment (terryttsai) [inline README.md:5]: Should we also link to the justice_counts directory here?

Comment (lilidworkin) [reply README.md]: good call, updated!

**Review by terryttsai — APPROVED** [review]

**Comment by terryttsai** [comment]
looks good!

**Comment by mxosman** [comment]
LGTM too - thank you, Lili!

**Pull request merged**